### PR TITLE
fix: Add PublicHealthQA

### PR DESCRIPTION
**Pull request by xhluca**
This is a work in progress

## Checklist for adding MMTEB dataset

Reason for dataset addition:

- [x] I have tested that the dataset runs with the `mteb` package.
- [x] I have run the following models on the task (adding the results to the pr). These can be run using the `mteb -m {model_name} -t {task_name}` command.
  - [x] `sentence-transformers/paraphrase-multilingual-MiniLM-L12-v2`
  - [x] `intfloat/multilingual-e5-small`
- [x] I have checked that the performance is neither trivial (both models gain close to perfect scores) nor random (both models gain close to random scores).
- [x] If the dataset is too big (e.g. >2048 examples), considering using `self.stratified_subsampling() under dataset_transform()`
- [x] I have filled out the metadata object in the dataset file (find documentation on it [here](https://github.com/embeddings-benchmark/mteb/blob/main/docs/adding_a_dataset.md#2-creating-the-metadata-object)).
- [x] Run tests locally to make sure nothing is broken using `make test`. 
- [x] Run the formatter to format the code using `make lint`. 
- [x] I have added points for my submission to the [points folder](https://github.com/embeddings-benchmark/mteb/blob/main/docs/mmteb/points.md) using the PR number as the filename (e.g. `438.jsonl`).

## Notes

> I have tested that the dataset runs with the `mteb` package.

I have tested with the following code successfully:

```python
from mteb import MTEB

evaluation = MTEB(tasks=["PublicHealthQA"])
evaluation.load_tasks_data()
phqa = evaluation.tasks[0]

q = phqa.queries['spanish']['test']['Q21']
doc_id = list(phqa.relevant_docs['spanish']['test']['Q21'].keys())
answer = phqa.corpus['spanish']['test'][doc_id]

print("Query:", q)
print("Answer:", doc['text'])
```

> I have run the following models on the task

Ran (0fe8e62062f3c361c669fa008aafd419d9a30f36):
```
CUDA_VISIBLE_DEVICES=6 mteb  -t PublicHealthQA -m sentence-transformers/paraphrase-multilingual-MiniLM-L12-v2
CUDA_VISIBLE_DEVICES=7 mteb  -t PublicHealthQA -m intfloat/multilingual-e5-small
```

> I have checked that the performance is neither trivial (both models gain close to perfect scores) nor random (both models gain close to random scores).

Looking at the numbers in the JSON, it seems the task is not solved since NDCG@10 is ~0.5. As for randomness, it seems recall@k increases progressively from 1 to 20, at 100 it is near 100% which is due to the small corpus size.

### CODE REVIEW

**Comment by xhluca** [comment]
> Run tests locally to make sure nothing is broken using make test.

I tried running locally but ran out of memory. I'm not sure if my tests are set up correctly to be run. Is it okay to wait for the CI tests instead?

**Comment by isaac-chung** [comment]
Thanks for adding this! I've enabled CI for the tests.
For points, please check if these languages are already covered within retrieval. If yes, then it'll be 2 pts for the dataset. If not, then it'll be 2 + 4 * each _new_ language. Plus 2 points for each reviewer.

File: mteb/tasks/Retrieval/multilingual/PublicHealthQARetrieval.py [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+@misc {xing_han_lu_2024,
+	author       = { {Xing Han Lu} },
+	title        = { publichealth-qa (Revision 3b67b6b) },
+	year         = 2024,
+	url          = { https://huggingface.co/datasets/xhluca/publichealth-qa },
+	doi          = { 10.57967/hf/2247 },
+	publisher    = { Hugging Face }
+}
+""",
+        n_samples=888,
```
Comment (wissam-sib) [inline mteb/tasks/Retrieval/multilingual/PublicHealthQARetrieval.py:96]:
```suggestion
        n_samples= {"test": 888},
```

File: mteb/tasks/Retrieval/multilingual/PublicHealthQARetrieval.py [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+        avg_character_length=778.1655,
```
Comment (wissam-sib) [inline mteb/tasks/Retrieval/multilingual/PublicHealthQARetrieval.py:97]:
```suggestion
        avg_character_length= {"test": 778.1655},
```

Comment (wissam-sib) [reply mteb/tasks/Retrieval/multilingual/PublicHealthQARetrieval.py:97]: This causes the tests to fail

Comment (xhluca) [reply mteb/tasks/Retrieval/multilingual/PublicHealthQARetrieval.py:97]: Thanks! Merging now.

**Comment by xhluca** [comment]
@isaac-chung where can I find a list of all languages in the task?

**Comment by isaac-chung** [comment]
I just took a look over the tasks under the `Retrieval` folder. All languages have been covered before. So this would be 2 pts for the dataset.

File: mteb/tasks/Retrieval/multilingual/PublicHealthQARetrieval.py [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+            "revision": "main",
+        },
+        type="Retrieval",
+        category="s2p",
+        eval_splits=[_EVAL_SPLIT],
+        eval_langs=_LANGS,
+        main_score="ndcg_at_10",
+        reference="https://huggingface.co/datasets/xhluca/publichealth-qa",
+        date=("2020-01-01", "2020-04-15"),
+        form="written",
```
Comment (isaac-chung) [inline mteb/tasks/Retrieval/multilingual/PublicHealthQARetrieval.py:78]:
```suggestion
        form=["written"],
```

**Review by isaac-chung — COMMENTED** [review]
Small comment. Good to add points.

**Comment by xhluca** [comment]
@isaac-chung added your comment and the points! Unfortunately the tests seem to be broken, as it keeps running out of memory and I can't restrict the number of workers (it always takes all cores).

**Comment by xhluca** [comment]
FYI the CI tests are failing because of unrelated issues:

```
============================= slowest 5 durations ==============================
17.24s call     tests/test_mteb.py::test_mteb_task[average_word_embeddings_levy_dependency-TwentyNewsgroupsClustering]
9.50s call     tests/test_mteb.py::test_two_mteb_tasks
9.42s call     tests/test_mteb_rerank.py::test_mteb_rerank
9.35s call     tests/test_mteb.py::test_mteb_task[average_word_embeddings_levy_dependency-SciDocsRR]
8.97s call     tests/test_mteb.py::test_mteb_task[average_word_embeddings_levy_dependency-Banking77Classification]
=========================== short test summary info ============================
FAILED tests/test_all_abstasks.py::test_dataset_availability - AssertionError: Datasets not available on Hugging Face:

```

**Comment by wissam-sib** [comment]
> FYI the CI tests are failing because of unrelated issues:
> 
> ```
> ============================= slowest 5 durations ==============================
> 17.24s call     tests/test_mteb.py::test_mteb_task[average_word_embeddings_levy_dependency-TwentyNewsgroupsClustering]
> 9.50s call     tests/test_mteb.py::test_two_mteb_tasks
> 9.42s call     tests/test_mteb_rerank.py::test_mteb_rerank
> 9.35s call     tests/test_mteb.py::test_mteb_task[average_word_embeddings_levy_dependency-SciDocsRR]
> 8.97s call     tests/test_mteb.py::test_mteb_task[average_word_embeddings_levy_dependency-Banking77Classification]
> =========================== short test summary info ============================
> FAILED tests/test_all_abstasks.py::test_dataset_availability - AssertionError: Datasets not available on Hugging Face:
> ```

Yes it was happening with all PRs, fixed it here: https://github.com/embeddings-benchmark/mteb/pull/761

**Pull request merged**